packages/block-library/src/image/media-upload-progress.native.js

```diff
@@ -0,0 +1,151 @@
+/**
```

Comment:
This is a new component to be used by both video and image blocks(and probably by other media upload blocks in the future). It is actually not a new code, just extracted from image block. 

It listens to media upload events coming from main apps and updates progress Spinner. It also provides a way to render a child component with the current upload state(see props.renderContent) If provided a coverUrl via props then it will also calculate width, height of it and pass to renderContent.

Reply:
Maybe a better place for this component would be `media-upload/`? Since it's meant to be shared between media related blocks.
It also looks like a good candidate for a HOC (not 100% sure). This might also make easier its implementation in blocks.

Reply:
Thanks @etoledom 👍 I added a tech debt here https://github.com/wordpress-mobile/gutenberg-mobile/issues/687#issuecomment-482011555